aodntools/timeseries_products/gridded_timeseries.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import bisect
+import argparse
+import os.path
+import json
+from datetime import datetime
+
+import xarray as xr
+import pandas as pd
+
+import aggregated_timeseries as TStools
+
+
+TEMPLATE_JSON = 'gridded_timeseries_template.json'
+
+
+def make_depth_bins(nc, increment=10):
+    """
+    generate a list of depth bins from min/max DEPTH.
+    Re-scale the min/max to nearest increment round number and start the bin from there
+    :param nc: hourly dataset
+    :return: list of depth bins
+    """
+
+    dmin = int(nc.DEPTH.min() / increment) * increment
+    dmax = int(nc.DEPTH.max() / increment) * increment + increment
+    dbin = list(range(dmin, dmax, increment))
+
+    return dbin
+
+def adjust_depth_bins(depth_bins, depth_min, depth_max):
+    """
+    Adjust the provided depth bins to match the min and max registered depth
+    :param depth_bins: list of target depths
+    :param depth_min: min valid depth
+    :param depth_max: max valid depth
+    :return: list of adjusted target depths
+    """
+
+    target_depth_min = bisect.bisect_left(depth_bins, depth_min)
+    target_depth_max = bisect.bisect_right(depth_bins, depth_max)
+    return depth_bins[target_depth_min:target_depth_max]
+
+
+def get_depth_mask(depth_bins, depths, max_separation):
+    """
+    return a boolean mask of depth bins where the interpolation is possible
+    based on maximum separation between the instruments
+    :param depth_bins: list of depth bins where the interpolation will happen
+    :param depths: list of depth of the instruments at the current timestamp
+    :param max_separation: maximum allowable separation between readings
+    :return: list of depth masked
+    """
+    depth_mask = [False] * len(depth_bins)
+    for depth_index in range(len(depths) - 1):
+        if depths[depth_index + 1] - depths[depth_index] < max_separation:
+            depth_min = bisect.bisect_left(depth_bins, depths[depth_index])
+            depth_max = bisect.bisect_right(depth_bins, depths[depth_index + 1])
+            depth_interval_len = depth_max - depth_min
+            depth_mask[depth_min:depth_max] = [True] * depth_interval_len
+    return depth_mask
+
+def sort_attributes(attributes):
```

Comment:
I don't see any logic that requires a sorted dictionary - So why do it? If it's for the final file to have global attributes in order, you can do at write time.

Reply:
yes, is to have the global attrs sorted in the output file. the sort in now done just before writing without the function